### ISSUE TITLE
Meaning of "Disabled as broken perl installs give false negative" warning in test suite

### ISSUE THREAD

**Issue by gregory-nisbet** [Perl-Toolchain-Gang/ExtUtils-MakeMaker#289]
This might be a documentation-level question, since it's about a specific warning message when a test is skipped.

I'm using Perl 5.24 (installed through Homebrew) on OS X 10.12.3 (Sierra).

I recently upgraded all my CPAN modules, and saw this warning in the test suite for `ExtUtils::MakeMaker`.

Here's the file where the warning was emitted.

https://github.com/Perl-Toolchain-Gang/ExtUtils-MakeMaker/blob/1b8f5daa8116d469b280b12b04c1e6f6af36f6c5/t/03-xsstatic.t

I'm trying to find out what's wrong with my environment that's preventing the test from running.

I installed Perl 5.24 through Homebrew (the community os x package manager). It's dynamically linked and links against `libperl.dylib`. I've never had a problem with it though.

I'm guessing based on this line that the `static perl` means statically linked perl. Here's an excerpt of the test for context.

```
plan skip_all => "Disabled as broken perl installs give false negative"
  # if not static perl, and not author
  unless !$Config{usedl} or $ENV{AUTHOR_TESTING};
```

Here are the libraries that the homebrew perl depends on.

```
% otool -L /usr/local/bin/perl
/usr/local/bin/perl:
	/usr/local/Cellar/perl/5.24.0_1/lib/perl5/5.24.0/darwin-thread-multi-2level/CORE/libperl.dylib (compatibility version 5.24.0, current version 5.24.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.0.0)
	/usr/lib/libutil.dylib (compatibility version 1.0.0, current version 1.0.0)
```

For comparison, here's the output for the system perl.

```
% otool -L /usr/bin/perl
/usr/bin/perl:
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1346.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.0.0)
```

Is there a specific reason why `perl` executables that link against `libperl` are considered broken? If there is, I can try to submit a patch to the homebrew formula.

**Comment by Leont** [comment]
I guess the message could be clearer. It's not perl itself that's broken.

If you have any module installed that leaves behind static libraries in the install directories (a few modules do that), that made subsequent MakeMaker upgrades fail this test, hence it's disabled.

**Comment by Leont** [comment]
This was fixed by #293
